### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by ngoiz**
A bit of a mixed PR. No major changes to the code, simply formatting changes, documentation and, hopefully, a useful algebra function.

Summary:
* Get desired projection matrix from its name in `algebra.py`
* Removed `StabilityDerivatives` post-proc. It is buggy and not production ready. Back to the development phase.
* Give the `SaveData` post-proc a default output folder `savedata` within the case output
* Add support for saving linearised system to `.h5` as well, previously only `.mat`.
* Minor formatting changes + documentation

EDIT:
The way settings for `DynamicControlSurface` generator are parsed in `AerogridLoader` have also changed. Instead of the settings `control_surface_deflection_generator_settings` taking in a list of dictionaries containing the settings for each control surface as before, we now take in a dictionary of dictionaries, where the key of the dictionary is the index of the control surface in string format, and the value is the dictionary of settings. Thus, the input may now look like this:
```
cs0_settings = {}  # these are the settings for the desired control surface
settings = {}
settings['AerogridLoader] = {'control_surface_deflection' : ['DynamicControlSurface],
                                                 'control_surface_deflection_generator_settings: {'0': cs_settings},
                                                }
```

### CODE REVIEW

**Comment by ngoiz** [comment]
This is ready for review. Have a look at the new function in `algebra` to get projection matrices and let me know if you think this makes sense or there is a better way to do this.

Also, if you have minor changes lying around feel free to include them in this PR.

File: sharpy/generators/dynamiccontrolsurface.py [hunk 1]
```diff
@@ -2,66 +2,85 @@
...
+    A first order finite-differences scheme is used to calculate the deflection rate based on the provided time step
+    increment.
 
-    The object generates a deflection in radians based on the time series given as a vector in the input data
+    To call this generator, the ``generator_id = DynamicControlSurface`` key shall be used for the setting
+    `control_surface_deflection` in the ``AerogridLoader`` solver.
 
-    To call this generator, the ``generator_id = DynamicControlSurface`` shall be used.
-    This is parsed as the value for the ``control_surface_deflection_generator`` key in the aerogridloader solver's settings.
+    These generator settings should be parsed as a dictionary to the setting
```
Comment (ArturoMS13) [inline sharpy/generators/dynamiccontrolsurface.py:20]:
```suggestion
    This generator settings should be parsed as a dictionary to the setting
```

File: sharpy/generators/dynamiccontrolsurface.py [hunk 2]
```diff
@@ -2,66 +2,85 @@
...
+    ``control_surface_deflection_generator_settings`` in ``AerogridLoader``, where this dictionary should be the value
+    to another dictionary where the key is the index of the control surface in string format. This is shown better
+    in the example below:
```
Comment (ArturoMS13) [inline sharpy/generators/dynamiccontrolsurface.py:23]:
```suggestion
   One instance of this generator will be created for each control surface, thus, a group of settings should be defined 
   for each control surface (``cs0_settings``, ``cs1_settings`` ... in the example below).
   All of these groups of settings should be collected as values in a dictionary which keys are the associated control surface number in string format. 
   This dictionary should be parsed to the variable
    ``control_surface_deflection_generator_settings`` in ``AerogridLoader``. This is shown better
    in the example below:
```

I think it is more clear like this, but feel free to discard this comment

File: sharpy/generators/dynamiccontrolsurface.py [hunk 3]
```diff
@@ -2,66 +2,85 @@
...
+
+        .. code-block:
 
-    Args:
-        in_dict (dict): Input data in the form of dictionary. See acceptable entries below.
+            cs0_settings = {}  # these are the settings for the desired control surface
+            settings = {}
+            settings['AerogridLoader] = {'control_surface_deflection' : ['DynamicControlSurface],
+                                         'control_surface_deflection_generator_settings: {'0': cs_settings},
+                                         }
```
Comment (ArturoMS13) [inline sharpy/generators/dynamiccontrolsurface.py:33]:
```suggestion
            cs0_settings = {}  # these are the settings for control surface number 0
            cs1_settings = {}  # these are the settings for control surface number 1
            dict_of_cs = {'0': cs0_settings,
                                  '1': cs1_settings} # This dictionary groups all the settings for all the control surfaces
            settings = {}
            settings['AerogridLoader] = {'control_surface_deflection' : ['DynamicControlSurface],
                                                         'control_surface_deflection_generator_settings: dict_of_cs}
```

Same as the above comment

File: sharpy/generators/dynamiccontrolsurface.py [hunk 4]
```diff
@@ -2,66 +2,85 @@
...
 
-            ======================   ===============  ======================================================================  ===================
-            Name                     Type             Description                                                             Default
-            ======================   ===============  ======================================================================  ===================
-            ``dt``                   ``float``        Timestep for the simulation                                             ``None``
-            ``deflection_file``      ``str``          Relative path to the file with the deflection information.              ``None``
-            ======================   ===============  ======================================================================  ===================
+
+    This is parsed as the value for the ``control_surface_deflection_generator`` key in the aerogridloader solver's settings.
```
Comment (ArturoMS13) [inline sharpy/generators/dynamiccontrolsurface.py:36]:
```suggestion
```

File: sharpy/postproc/aeroforcescalculator.py [hunk 5]
```diff
@@ -96,7 +92,7 @@ def run(self, online=False):
             # create folder for containing files if necessary
             if not os.path.exists(self.folder):
                 os.makedirs(self.folder)
-            self.folder += self.settings['text_file_name']
+            self.folder += 'aeroforces.txt'
```
Comment (ArturoMS13) [inline sharpy/postproc/aeroforcescalculator.py:95]: This change looks like a loss of functionalities, is there a reason for it?

File: sharpy/postproc/savedata.py [hunk 6]
```diff
@@ -137,7 +137,7 @@ def initialise(self, data, custom_settings=None, caller=None):
         # create folder for containing files if necessary
         if not os.path.exists(self.settings['folder']):
             os.makedirs(self.settings['folder'])
-        self.folder = self.settings['folder'] + '/' + self.data.settings['SHARPy']['case'] + '/'
+        self.folder = self.settings['folder'] + '/' + self.data.settings['SHARPy']['case'] + '/savedata/'
         if not os.path.exists(self.folder):
             os.makedirs(self.folder)
         self.filename = self.folder + self.data.settings['SHARPy']['case'] + '.data.h5'
```
Comment (ArturoMS13) [inline sharpy/postproc/savedata.py:143]:
```suggestion
        self.filename = self.folder + self.data.settings['SHARPy']['case'] + '.data.h5'
        self.filename_linear = self.folder + self.data.settings['SHARPy']['case'] + '.linss.h5'
```

File: sharpy/postproc/savedata.py [hunk 7]
```diff
@@ -233,6 +233,18 @@ def run(self, online=False):
                                    compress_float=self.settings['compress_float'])
                 linhdffile.close()
 
+            if self.settings['save_linear']:
+                with h5py.File(self.filename.replace('.data.h5', '.linss.h5'), 'a') as linfile:
```
Comment (ArturoMS13) [inline sharpy/postproc/savedata.py:237]:
```suggestion
                with h5py.File(self.filename_linear, 'a') as linfile:
```

File: sharpy/postproc/savedata.py (hunk 6 above)
Comment (ArturoMS13) [inline sharpy/postproc/savedata.py:143]:
```suggestion
        self.filename = self.folder + self.data.settings['SHARPy']['case'] + '.data.h5'
        self.filename_linear = self.folder + self.data.settings['SHARPy']['case'] + '.linss.h5'
```

**Review by ArturoMS13 — CHANGES_REQUESTED** [review]
Thanks for the contribution. I just added a couple of comments but this is ready to merge

File: sharpy/postproc/aeroforcescalculator.py (hunk 5 above)
Comment (ngoiz) [reply sharpy/postproc/aeroforcescalculator.py]: I can't think of a good reason, so it's going back

**Comment by ngoiz** [comment]
Thanks for the comments. I have addressed them now.

Only missing the quick fix to the docs required by #109 and should be good to go.

**Comment by ngoiz** [comment]
Closed #109, so this should be good

**Review by ArturoMS13 — APPROVED** [review]
This looks ready

**Pull request merged**